src/components/Dropdown.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { useCallback, useMemo, useRef, useState } from 'react';
+import { Person } from '../types/Person';
+import classNames from 'classnames';
+import Notification from './Notification';
+
+import debounce from 'lodash.debounce';
+
+type Props = {
+  people: Person[];
+  onSelect: (person: Person | null) => void;
+};
+
+const Dropdown: React.FC<Props> = ({ people, onSelect }) => {
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [focused, setFocused] = React.useState(false);
+  const onFocus = () => setFocused(true);
+  const onBlur = () => setFocused(false);
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, 300), []);
+
+  const nameField = useRef<HTMLInputElement>(null);
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    applyQuery(event.target.value);
+    if (query !== '') {
```

Comment:
```suggestion
    if (query) {
```
You don't need to compare with an empty string